### PR TITLE
V1 fix filter by repo

### PR DESCRIPTION
**Pull request by isc-shuliu**
With this change, user can list installed modules from a certain repository. E.g.:

```
list -repo registry
```

### CODE REVIEW

**Comment by isc-tleavitt** [comment]
@isc-shuliu is there an issue associated with this?

**Comment by isc-shuliu** [comment]
> @isc-shuliu is there an issue associated with this?

@isc-tleavitt Yes, it's here https://github.com/intersystems/ipm/issues/510#issuecomment-2248004980

File: src/cls/IPM/Main.cls [hunk 1]
```diff
@@ -960,12 +961,16 @@ ClassMethod GetListModules(pNamespace As %String = {$Namespace}, pSearch As %Str
...
-      Set tQuery = tQuery _ " WHERE name IN (" _ $ListToString(tParams, ", ") _ ")"
+      Set tQuery = tQuery _ " AND name IN (" _ $ListToString(tParams, ", ") _ ")"
     } ElseIf pSearch'="*" {
-      Set tQuery = tQuery _ " WHERE name " _ $Select(pSearch["*": "like ?", 1: "= ?")
+      Set tQuery = tQuery _ " AND name " _ $Select(pSearch["*": "like ?", 1: "= ?")
       Set tArgs($Increment(tArgs)) = $Translate(pSearch, "*", "%")
     }
   }
+  If pRepository'="" {
+	Set tQuery = tQuery _ " AND Repository = ?"
```
Comment (isc-kiyer) [inline src/cls/IPM/Main.cls:971]: Nit: Fix spacing of contents inside if block

File: src/cls/IPM/Main.cls [hunk 2]
```diff
@@ -960,12 +961,16 @@ ClassMethod GetListModules(pNamespace As %String = {$Namespace}, pSearch As %Str
         Set tParams = tParams _ $ListBuild("?")
         Set tArgs($Increment(tArgs)) = tItem
       }
-      Set tQuery = tQuery _ " WHERE name IN (" _ $ListToString(tParams, ", ") _ ")"
+      Set tQuery = tQuery _ " AND name IN (" _ $ListToString(tParams, ", ") _ ")"
```
Comment (isc-tleavitt) [inline src/cls/IPM/Main.cls:964]: Note: you don't need to fix this, but all this code would be much cleaner if it was just name %INLIST ? where ? is the $ListBuild list from line 955.

**Review by isc-tleavitt — APPROVED** [review]
Seems like some of this was just missed in a merge somewhere in the past?

**Review by isc-kiyer — APPROVED** [review]
Looks good to me. One minor nit

File: src/cls/IPM/Main.cls (hunk 1 above)
Comment (isc-shuliu) [reply src/cls/IPM/Main.cls]: Done.

**Review by isc-kiyer — APPROVED** [review]

**Pull request merged**